src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs

```
@@ -319,7 +319,68 @@ public void Visit(TimeValue timeValue)
 
                 public void Visit(UntypedObjectValue untypedObjectValue)
                 {
-                    throw new ArgumentException($"Unable to serialize type {untypedObjectValue.GetType().FullName} to Json format.");
+                    Visit(untypedObjectValue.Impl);
+                }
+
+                private void Visit(IUntypedObject untypedObject, int index = 0)
+                {
+                    FormulaType type = untypedObject.Type;
+
+                    if (index > 40)
+                    {
+                        return;
```

>return [](http://example.com/codeflow?start=24&length=6)

We should return an error instead of silently not serializing what the user has